epoxy-adapter/src/test/java/com/airbnb/epoxy/EpoxyVisibilityTrackerTest.kt

```diff
@@ -0,0 +1,535 @@
+package com.airbnb.epoxy
+
+import android.app.Activity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.TextView
+import com.airbnb.epoxy.VisibilityState.FOCUSED_VISIBLE
+import com.airbnb.epoxy.VisibilityState.FULL_IMPRESSION_VISIBLE
+import com.airbnb.epoxy.VisibilityState.INVISIBLE
+import com.airbnb.epoxy.VisibilityState.UNFOCUSED_VISIBLE
+import com.airbnb.epoxy.VisibilityState.VISIBLE
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.annotation.Config
+import java.lang.StringBuilder
+
+/**
+ * This class test the EpoxyVisibilityTracker by using a RecyclerView that scroll vertically. The
+ * view port height is provided by Robolectric.
+ *
+ * We are just controlling how many items are displayed with VISIBLE_ITEMS constant.
+ */
+@Config(sdk = [21], manifest = TestRunner.MANIFEST_PATH)
+@RunWith(TestRunner::class)
+class EpoxyVisibilityTrackerTest {
```

Comment:
I just added unit tests (in Kotlin). @elihart please review and let me know what you think. I think theses are the first tests in the lib to assert on actual UI (`RecyclerView`). Btw, I was able to fix few bugs related to use case where `scrollToPosition(...)` is called.